### ISSUE TITLE
Add ability to provide a python object with load/loads

### ISSUE THREAD

**Issue by Grokzen** [ssato/python-anyconfig#16]
Hi.

I have a usecase and i have not found a solution in the docs or in the code yet. When loading config files i want to be able to provide some sane default values that is hardcoded into the application code and then load either some global config file paths or user specified files. This data is a nested python dict.

**Comment by ssato** [comment]
I'm now working on a similar enhancement before the next release 0.0.7 by chance.
Could you please wait a little? I'll commit that enhancement within a few hour hopefully.

**Comment by Grokzen** [comment]
Ye i am in no hurry :]

**Comment by Grokzen** [comment]
@ssato I was also thinking about something else. Is there some built in way right now to handle loading 2 config files at different times in my code but still have them merged?

The data that is returned from any of the load functions is the result dict that was read from the files. Or will this usecase also be improved in your next release somehow?

**Comment by ssato** [comment]
@Grokzen Maybe a code like the following accomplish that, I guess:

```python
import anyconfig

c0 = anyconfig.load("/path/to/config_0.yml")  # c0 :: anyconfg.mergeabledict.MergeableDict
    ...
c1 = anyconfig.load("/path/to/config_1.yml")
c0.update(c1)  # or c0.update(c1, anyconfig.MS_XXXXX)
```

On the other hand, what I'm now working on is to support to pass config object
(dict or dict-like object) to anyconfig.\*load\* to provide default config values.

**Comment by ssato** [comment]
I struggled with my idea for a while conlusion is that it brings another complexity
So, I quit the work.

Instead, I add a note to show how to set default values. Could you please take
a look at the "Default config values" section in README.rst ?

**Comment by Grokzen** [comment]
I tried the suggestion you written but i think it is abit bugged or wrong in the case where i in the loaded config file had a new key that was not already set in the defauls.

```python
In [7]: conf.update(**conf_from_files)
TypeError: update() got an unexpected keyword argument 'a'
update() got an unexpected keyword argument 'a'
```

But the following updated the first MergableDict properly

```python
In [11]: conf.update(conf_from_files)
In [12]: print(conf)
{'a': {'b': True}, 'b': True, 'bar': '1', 'foobar': True, 'baz': [2, 3], 'opa': 'AsdAsd', 'foo': 1}
```

So i guess you can't use `**` with the update() method.

Also in the example of loading the default values i would have written because i would probably use a predefined dict to set any default values.

```python
conf = anyconfig.container({'foo': 0, 'bar': '1', 'baz': [2, 3]})
```

**Comment by ssato** [comment]
You're right. '**' was not needed. I fix it.

**Comment by ssato** [comment]
I've just released 0.0.7 contains the workaround for this issue and I close this.
